lib/crypto/crypto_keys.c

```diff
@@ -7,6 +7,7 @@
 
 #include <openssl/crypto.h>
 #include <openssl/err.h>
+#include <openssl/ossl_typ.h>
```

Comment:
This seems like an internal OpenSSL header.  The RSA_* functions are documented only as needing <openssl/rsa.h>, and in fact that's guaranteed to provide the type definition we need since it provides functions which declare the type.